### ISSUE TITLE
App crashes on opening large .txt file

### ISSUE THREAD

**Issue by ghost** [soupslurpr/BeauTyXT#140]
app works fine normally, I think the cause is a greater than 2mb text file. I have replicated every single time after trying to load my logs from an app called mutiny wallet, which end up being a 2.1mb text file of logs. timeout is attached here:

```
type: ANR
osVersion: google/husky/husky:14/AP1A.240405.002/2024040900:user/release-keys
package: dev.soupslurpr.beautyxt:53
process: dev.soupslurpr.beautyxt
installer: app.accrescent.client

activity: dev.soupslurpr.beautyxt/.MainActivity
cause: ANR Input dispatching timed out (b9c7dc2 dev.soupslurpr.beautyxt/dev.soupslurpr.beautyxt.MainActivity (server) is not responding. Waited 5003ms for FocusEvent(hasFocus=true)).
info: ANR in dev.soupslurpr.beautyxt (dev.soupslurpr.beautyxt/.MainActivity)
PID: 7618
Reason: Input dispatching timed out (b9c7dc2 dev.soupslurpr.beautyxt/dev.soupslurpr.beautyxt.MainActivity (server) is not responding. Waited 5003ms for FocusEvent(hasFocus=true)).
Parent: dev.soupslurpr.beautyxt/.MainActivity
ErrorId: 9efaa058-7862-4735-a89e-bad1b4487da1
Frozen: false
Load: 3.72 / 3.71 / 4.13
----- Output from /proc/pressure/memory -----
some avg10=0.03 avg60=0.41 avg300=0.78 total=47939472
full avg10=0.03 avg60=0.37 avg300=0.69 total=41213657
----- End output from /proc/pressure/memory -----
----- Output from /proc/pressure/cpu -----
some avg10=6.80 avg60=8.69 avg300=10.17 total=259211123
full avg10=0.00 avg60=0.00 avg300=0.00 total=0
----- End output from /proc/pressure/cpu -----
----- Output from /proc/pressure/io -----
some avg10=0.03 avg60=0.10 avg300=0.13 total=18396921
full avg10=0.02 avg60=0.06 avg300=0.06 total=11127745
----- End output from /proc/pressure/io -----
```

![Screenshot_20240420-044421.png](https://github.com/soupslurpr/BeauTyXT/assets/114684096/6b3c9191-4a82-4b9d-b90e-be6f8a5363ee)

it just gets stuck on this screen then android says "beautyxt" isn't responding"

**Comment by soupslurpr** [comment]
I'm fairly certain the issue is with us trying to render all that text at once. The compose TextField isn't optimized rn for that and it just renders everything inside all at once even if it is off screen it seems. So unfortunately it's somewhat upstream since I don't think there are other solutions for that.

**Comment by soupslurpr** [comment]
For just viewing it, try a Chromium based browser for now and it should work fine :)

**Comment by soupslurpr** [comment]
> I'm fairly certain the issue is with us trying to render all that text at once. The compose TextField isn't optimized rn for that and it just renders everything inside all at once even if it is off screen it seems. So unfortunately it's somewhat upstream since I don't think there are other solutions for that.

You can see this when trying to load other large files the UI is laggy.

**Comment by bedair81** [comment]
Yeah that's what I ended up doing, just wanted to make the issue known since I didn't know the cause.

appreciate the help mate :)

**Comment by soupslurpr** [comment]
> [!NOTE]
> GPT-6-Astra, responding on behalf of soupslurpr.

[BeauTyXT Reimagined (version 84)](https://github.com/soupslurpr/BeauTyXT/releases/tag/84) is now released. It requires Android 17 / API 37 or newer.

Large-file loading and editing were rewritten around a Rust document engine and bounded source windows, avoiding the old full-file Compose TextField path discussed above. The release supports UTF-8 files up to 256 MiB and has passed 100 MiB workflow checks. Closing this old large-file ANR report as addressed. If an ANR still occurs in version 84, please comment with the device/version and, if possible, a non-sensitive reproducer.
